web-bundle/src/main/java/com/graphhopper/resources/MapboxResponseConverter.java

```diff
@@ -0,0 +1,324 @@
+        }
+
+        final ArrayNode waypointsJson = json.putArray("waypoints");
+        for (int i = 0; i < waypoints.size(); i++) {
+            ObjectNode waypointJson = waypointsJson.addObject();
+            // TODO get names
+            waypointJson.put("name", "");
+            putLocation(waypoints.getLat(i), waypoints.getLon(i), waypointJson);
+        }
+
+        json.put("code", "Ok");
+        // TODO: Maybe we need a different format... uuid: "cji4ja4f8004o6xrsta8w4p4h"
+        json.put("uuid", UUID.randomUUID().toString().replaceAll("-", ""));
+
+        return json;
+    }
+
+    private static ObjectNode putInstruction(InstructionList instructions, int index, Locale locale, ObjectNode instructionJson) {
+        Instruction instruction = instructions.get(index);
+        ArrayNode intersections = instructionJson.putArray("intersections");
+        ObjectNode intersection = intersections.addObject();
+        intersection.put("out", 0);
+        intersection.putArray("entry");
+        intersection.putArray("bearings");
+        PointList pointList = instruction.getPoints();
+        putLocation(pointList.getLat(0), pointList.getLon(0), intersection);
+
+        instructionJson.put("driving_side", "right");
+
+        // Does not include elevation
+        instructionJson.put("geometry", WebHelper.encodePolyline(pointList, false, 1e6));
+
+        // TODO: how about other modes?
+        instructionJson.put("mode", "driving");
+
+        putManeuver(instruction, instructionJson, index, locale);
+
+        // TODO distance = weight, is weight even important?
+        double distance = Helper.round(instruction.getDistance(), 10);
+        instructionJson.put("weight", distance);
+        instructionJson.put("duration", convertToSeconds(instruction.getTime()));
+        instructionJson.put("name", instruction.getName());
+        instructionJson.put("distance", distance);
+
+        ArrayNode voiceInstructions = instructionJson.putArray("voiceInstructions");
+        ArrayNode bannerInstructions = instructionJson.putArray("bannerInstructions");
+
+        // Voice and banner instructions are empty for the last element
+        if (index + 1 < instructions.size()) {
+            putVoiceInstructions(instructions, distance, index, locale, voiceInstructions);
+            putBannerInstruction(instructions, distance, index, locale, bannerInstructions);
+        }
+
+        return instructionJson;
+    }
+
+    private static void putVoiceInstructions(InstructionList instructions, double distance, int index, Locale locale, ArrayNode voiceInstructions) {
+        /*
+            A VoiceInstruction Object looks like this
+            {
+                distanceAlongGeometry: 40.9,
+                announcement: "Exit the traffic circle",
+                ssmlAnnouncement: "<speak><amazon:effect name="drc"><prosody rate="1.08">Exit the traffic circle</prosody></amazon:effect></speak>",
```

Comment:
I love this xml style inside JSON :D

Reply:
This is actually called [SSML](https://en.wikipedia.org/wiki/Speech_Synthesis_Markup_Language) which is used to change the way TTS speaks the instructions.